### ISSUE TITLE
Valid or Same padding in Conv2D?

### ISSUE THREAD

**Issue by marintoro** [Kaixhin/Rainbow#34]
Hello,

I just realize by looking at the code of the open-source Rainbow agent from Google ([here](https://github.com/google/dopamine)) that the network they use is using "Same" padding (the default in Tensorflow) while in torch the default is to use "Valid" padding.

This result in a really different network, indeed in the Tensorflow case, the first fully connected is taking a 7744 size vector (after flatten, it's [11,11,64] before flatten) while in your case using PyTorch the same fully connected is taking a 3136 size vector (it's [7,7,64] before flatten).

Don't know if it's really a problem but I thought it was interesting to notice...

**Comment by Kaixhin** [comment]
Dopamine was created by Brain, not DeepMind, so I think they got that detail wrong (if they were trying to replicate the original Nature network). Source code for the Nature network is [here](https://github.com/deepmind/dqn/blob/master/dqn/convnet.lua), and by default Torch7's 2D convolutions have [no padding](https://github.com/torch/nn/blob/master/SpatialConvolution.lua).
